### DISCUSSION TITLE
SIP-7 Specify error type for substandard validation failures

### DISCUSSION THREAD

**Discussion by frankisawesome** [ProjectOpenSea/SIPs#15] [SIP Discussions]
I think it makes sense for the SIP to specify the error type returned when the zone fails to validate a particular substandard check.

A good one could be `SubstandardViolation(uint8 substandardId, string reason, bytes32 orderHash`

**Comment by ryanio** [comment]
We already have `error InvalidSubstandard(uint8 substandard)` in [SIP-10](https://github.com/ProjectOpenSea/SIPs/blob/4c5dfb88c76464a61bf934bb1d1d7e622ecf2e3d/SIPS/sip-10.md?plain=1#L42). I like your idea but harcoding reasons might be too verbose and cost too much in terms of contract size (similar to old school string revert reasons compared to custom errors). So better custom errors might be a preferred approach. Adding orderHash seems smart too :)

**Reply by frankisawesome** [reply]
Yep fair enough - I will need to read up on SIP-10 and will close this now :)
